nailgun/entity_mixins.py

```diff
@@ -590,13 +590,33 @@ def __eq__(self, other):
         """
         if other is None:
             return False
+        return self.to_json_dict() == other.to_json_dict()
+
+    def compare(self, other, filter_fcn=None):
+        """Returns True if properties can be compared in terms of eq.
+        Entity's Fields can be filtered accordingly to 'filter_fcn'.
+        This callable receives field's name as first parameter and field itself
+        as second parameter.
+        It must return True if field's value should be included on
+        comparison and False otherwise. If not provided field's marked as
+        unique will not be compared by default. 'id' and 'name' are examples of
+        unique fields commonly ignored. Check Entities fields for fields marked
+        with 'unique=True'
+
+
+        :param other: entity to compare
+        :param filter_fcn: callable
+        :return: boolean
+        """
+        if other is None:
```

Comment:
this comment apply to  __eq__ also, as we should have an exception that say that no attr  to_json_dic, instead directly return False